packages/2d/src/Components/Geometry.ts

```diff
@@ -38,6 +41,21 @@ function Geometry<S extends Shape>({
     },
   };
 
+  const { onMouseDown } = useNewComponent(LowLevelMouse);
+
+  onMouseDown(({ pos }) => {
+    if (!shape.containsPoint(pos)) {
+      return;
+    }
+
+    const worldPos = useEntityTransforms()
+      .matrixForWorldPosition()
+      .transformPoint(pos);
+
+    const ent = useEntitiesAtPoint(worldPos)[0];
+    useInspectorSelectEntity(() => ent);
```

Comment:
We could guard this to only call `useInspectorSelectEntity` if `ent === useEntity()`; that way, we won't call it more than once

Reply:
Good point! Added.